src/components/companies/edit/CompanyEditDetails.tsx

```diff
@@ -0,0 +1,305 @@
+import { useState, useEffect } from "react";
+import { validateValue } from "../../../utils/editorValidation";
+import type {
+  CompanyDetails as CompanyDetailsType,
+  GicsOption,
+} from "@/types/company";
+import { IconCheckbox } from "@/components/ui/icon-checkbox";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectTrigger,
+  SelectValue,
+  SelectContent,
+  SelectItem,
+} from "@/components/ui/select";
+import { Undo2 } from "lucide-react";
+import { useTranslation } from "react-i18next";
+import { useGicsCodes } from "@/hooks/companies/useGicsCodes";
+import { useCompanyEditDetailsSave } from "@/hooks/companies/useCompanyEditDetailsSave";
+
+export function CompanyEditDetails({
+  company,
+  onSave,
+}: {
+  company: CompanyDetailsType;
+  onSave?: () => void;
+}) {
+  const { t } = useTranslation();
+  const [subIndustryCode, setSubIndustryCode] = useState(
+    company.industry?.industryGics?.subIndustryCode
+      ? String(company.industry.industryGics.subIndustryCode)
+      : "",
+  );
+  const [industryVerified, setIndustryVerified] = useState(
+    !!company.industry?.metadata?.verifiedBy,
+  );
+  const [baseYear, setBaseYear] = useState(company.baseYear?.year || "");
+  const [baseYearVerified, setBaseYearVerified] = useState(
+    !!company.baseYear?.metadata?.verifiedBy,
+  );
+  const [comment, setComment] = useState<string>("");
+  const [source, setSource] = useState<string>("");
+  const {
+    data: gicsOptions = [],
+    isLoading: gicsLoading,
+    isError: gicsIsError,
+    error: gicsErrorObj,
+  } = useGicsCodes();
+  const gicsError = gicsIsError
+    ? gicsErrorObj instanceof Error
+      ? gicsErrorObj.message
+      : "Failed to load industry options"
+    : null;
+  const [error, setError] = useState<string | null>(null);
+  const {
+    saveCompanyEditDetails,
+    isPending: loading,
+    error: mutationError,
+  } = useCompanyEditDetailsSave();
+
+  // Reset to false if value changes
+  useEffect(() => {
+    setIndustryVerified(
+      subIndustryCode ===
+        (company.industry?.industryGics?.subIndustryCode || "")
+        ? !!company.industry?.metadata?.verifiedBy
+        : false,
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [subIndustryCode]);
+
+  useEffect(() => {
+    setBaseYearVerified(
+      String(baseYear) === String(company.baseYear?.year || "")
+        ? !!company.baseYear?.metadata?.verifiedBy
+        : false,
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [baseYear]);
```

Comment:
Curious why the check is disabled here instead of adding the dependencies?

Reply:
I think as a rule of thumb we should avoid silencing the warning, even if we think that it will not matter since it will silence the warning for cases when it does matter as well. These bugs are generally quite hard to find since it will not show up as errors anywhere, only that we are operating on obsolete values for some cases. 

It could also be a sign that the effect should be split up, if we only care about certain values changing, so that we have one effect that works on that and updates a value which in turn triggers another effect that also take the other values into account.